apps/judicial-system/backend/src/app/modules/case/guards/rolesRules.ts

```diff
@@ -189,29 +195,39 @@ export const prosecutorTransitionRule: RolesRule = {
   dtoFieldValues: [
     CaseTransition.OPEN,
     CaseTransition.ASK_FOR_CONFIRMATION,
+    CaseTransition.DENY_INDICTMENT,
     CaseTransition.SUBMIT,
     CaseTransition.ASK_FOR_CANCELLATION,
     CaseTransition.DELETE,
     CaseTransition.APPEAL,
     CaseTransition.WITHDRAW_APPEAL,
-    CaseTransition.DENY_INDICTMENT,
   ],
   canActivate: (request) => {
+    const user: User = request.user
+    const dto: TransitionCaseDto = request.body
```

Comment:
This improves type inference so that we do not inadvertently access non-existent fields.